themes/bootstrap3/js/lightbox.js

```diff
@@ -131,21 +131,22 @@ VuFind.register('lightbox', function Lightbox() {
           VuFind.refreshPage();
           return;
         }
+        var testDiv = $('<div/>').html(content);
+        var errorMsgs = testDiv.find('.flash-message.alert-danger:not([data-lightbox-ignore])');
         // Place Hold error isolation
-        if (obj.url.match(/\/Record/) && (obj.url.match(/Hold\?/) || obj.url.match(/Request\?/))) {
-          var testDiv = $('<div/>').html(content);
-          var error = testDiv.find('.flash-message.alert-danger');
-          if (error.length && testDiv.find('.record').length) {
-            showAlert(error[0].innerHTML, 'danger');
+        if (obj.url.match(/\/(?!Record|Hold|Request)\?/)) {
+          if (errorMsgs.length && testDiv.find('.record').length) {
+            showAlert(errorMsgs[0].innerHTML, 'danger');
             return false;
           }
         }
         if ( // Close the lightbox after deliberate login
-          obj.method                                                                 // is a form
-          && ((obj.url.match(/MyResearch/) && !obj.url.match(/Bulk/) && !obj.url.match(/Delete/) && !obj.url.match(/Recover/)) // that matches login/create account
-            || obj.url.match(/catalogLogin/))                                        // or catalog login for holds
-          && $('<div/>').html(content).find('.flash-message.alert-danger').length === 0 // skip failed logins
+          obj.method && (                                         // is a form
+            obj.url.match(/catalogLogin/) ||                      // catalog login for holds
+            obj.url.match(/MyResearch\/(?!Bulk|Delete|Recover)/)  // or that matches login/create account
```

Comment:
For consistency with PHP style, would it be better to put || at the start of the second line rather than the end of the first? I've gotten used to looking at the front of lines for operators, though if you feel this way is better for JS, I can live with it.